File: examples/todolist/event/item_created.go
package event

type ItemCreated struct {
	Id     string `json:"id"`
	ListId string `json:"listId"`
	Text   string `json:"text"`
}

func (e ItemCreated) Tags() []string {
	return []string{
		ListTagPrefix(e.ListId),
		ItemTagPrefix(e.Id),
	}
}

File: fairway_ex/lib/examples/todolist/event/item_created.ex
defmodule Todolist.Event.ItemCreated do
  @moduledoc "An item was added to a todo list."

  @behaviour Fairway.EventData
  @derive Jason.Encoder
  defstruct [:item_id, :list_id, :text]

  @impl true
  def tags(%{item_id: iid, list_id: lid}), do: ["list_id:#{lid}", "item_id:#{iid}"]

  @impl true
  def type_name, do: "Todolist.ItemCreated"
end
